api/lib/infrastructure/repositories/schooling-registration-repository.js

```diff
@@ -55,11 +56,24 @@ module.exports = {
     const trx = await Bookshelf.knex.transaction();
 
     try {
+      const nationalStudentIdsFromFile = schoolingRegistrationDatas.map((schoolingRegistrationData) => schoolingRegistrationData.nationalStudentId);
+      const students = await studentRepository.findReconciledStudentsByNationalStudentId(nationalStudentIdsFromFile);
 
       const schoolingRegistrations = _.map(schoolingRegistrationDatas, (schoolingRegistrationData) => new SchoolingRegistration({ ...schoolingRegistrationData, organizationId }));
       const schoolingRegistrationsFromOrganization = await this.findByOrganizationId({ organizationId });
-      const nationalStudentIdsFromOrganization = _.map(schoolingRegistrationsFromOrganization, 'nationalStudentId');
-      const [ schoolingRegistrationsToUpdate, schoolingRegistrationsToCreate ] = _.partition(schoolingRegistrations, (schoolingRegistration) => _.includes(nationalStudentIdsFromOrganization, schoolingRegistration.nationalStudentId));
+
+      const [ schoolingRegistrationsToUpdate, schoolingRegistrationsToCreate ] = _.partition(schoolingRegistrations, (schoolingRegistration) => {
+        const schoolingRegistrationFromOrganization = schoolingRegistrationsFromOrganization.find((schoolingRegistrationFromOrganization) => schoolingRegistrationFromOrganization.nationalStudentId === schoolingRegistration.nationalStudentId);
+        if (!schoolingRegistrationFromOrganization || (schoolingRegistrationFromOrganization && !schoolingRegistrationFromOrganization.userId)) {
+          if (students) {
+            const student = students.find((student) => student.nationalStudentId === schoolingRegistration.nationalStudentId);
+            if (student) {
+              schoolingRegistration.userId = student.getMostRelevantAccountForReconciliation().userId;
+            }
+          }
```

Comment:
L'appel à `const students = await studentRepository.findReconciledStudentsByNationalStudentId(nationalStudentIdsFromFile);` a lieu hors de la boucle, mais si on décale ici, on gagne en lisibilité (plus besoin de scroller pour savoir ce qu'il y a dans student).

```suggestion
         const students = await studentRepository.findReconciledStudentsByNationalStudentId(reconciliatingSchoolingRegistration.nationalStudentId);
          if (students) {
              schoolingRegistration.userId = student.getMostRelevantAccountForReconciliation().userId;
            }
          
```

Reply:
Tu exploses les performances en faisant ça

Reply:
Pas forcément, car ce n'est appelé que si l'appel suivant renvoie qqch
` schoolingRegistrationsFromOrganization.find((schoolingRegistrationFromOrganization) => schoolingRegistrationFromOrganization.nationalStudentId === schoolingRegistration.nationalStudentId);`

Faudrait tester et benchmark pour savoir